src/service/action-impl.js

```diff
@@ -176,7 +176,15 @@ export class ActionService {
       // fast-click.
       this.root_.addEventListener('click', event => {
         if (!event.defaultPrevented) {
-          this.trigger(dev().assertElement(event.target), 'tap', event);
+          this.trigger(dev().assertElement(event.target), name, event);
+        }
+      });
+      this.root_.addEventListener('keydown', event => {
+        if (event.keyCode == 13 /* enter */ ||
+            event.keyCode == 32 /* space */) {
+          if (!event.defaultPrevented) {
```

Comment:
You are correct, I was wrongly assuming `document.activeElement` is null initially, but it is `body` so no point in adding this.

Thinking about this more, I am concerned it will introduce unexpected behaviour for some elements. So I am suggesting we restrict this to only targets that have `role=button` attribute. What I am worried about is 1- How `inputs` that have `on=tap` might be handled (since space and enter have very different meanings there 2- How about elements that already have default behaviour for the keyboard like buttons/links. Would `tap` function be triggered twice? 3- Some elements like `<a>` handle `enter` but not `space` by design.

If we restrict to elements with `role=button` that handles the vast majority of the cases that are currently inaccessible (e.g. use of `div` as `button`). Fortunately our validator ensures anything that has `on:tap` also has a `role` and `tabindex` or they won't be valid AMP.